monkey/tests/unit_tests/monkey_island/cc/services/authentication_service/test_authentication_service.py

```diff
@@ -62,25 +71,28 @@ def authentication_facade(
     mock_user_datastore: UserDatastore,
     mock_token_generator: TokenGenerator,
     mock_token_validator: TokenValidator,
+    mock_token_parser: TokenParser,
 ) -> AuthenticationFacade:
     return AuthenticationFacade(
         mock_repository_encryptor,
         mock_island_event_queue,
         mock_user_datastore,
         mock_token_generator,
         mock_token_validator,
+        mock_token_parser,
     )
 
 
 def test_needs_registration__true(authentication_facade: AuthenticationFacade):
+    authentication_facade._datastore.find_user.return_value = False
     assert authentication_facade.needs_registration()
 
 
 def test_needs_registration__false(
     monkeypatch,
     authentication_facade: AuthenticationFacade,
 ):
-    User(username=USERNAME, password=PASSWORD).save()
+    authentication_facade._datastore.find_user.return_value = True
```

Comment:
Why is this change necessary? Doesn't `UserDatastore` rely on the `User` model? If we add or remove `Users` directly using the `User` model, shouldn't `UserDatastore` return the expected values?

I don't necessarily object to these changes, but I find it concerning that they need to be made.

Reply:
We're now using `UserDatastore` instead of the `User` model in the `needs_registration()` method. See 59d5c0707c441adf0f471b481a725a2d87998e3a. The datastore is mocked in these tests so the return value needs to be specified.